### ISSUE TITLE
BUG: 'cp -r' is not recognised as implementation defined

### ISSUE THREAD

**Issue by herpes-free-engineer-hpe** [koalaman/shellcheck#2302]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

cp -r a/ b/

#------------------------------------------------------------------------------

```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!
```
#### Here's what I wanted or expected to see:

```
Line 2:
cp -R a/ b/
    ^-- SC???: In POSIX sh, -r is undefined
```
See [Difference between cp -r and cp -R (copy command)](https://unix.stackexchange.com/a/18718/147432)

Thank you.

**Comment by brother** [comment]
I am no expert in this but -R is posix since 2008. No?

 ```
      Earlier	versions of this standard included support for the -r option to copy file hierarchies. The -r option is historical practice on BSD
       and BSD-derived systems. This option is no longer specified by POSIX.1-2008 but may be present in some implementations. The -R  option  was
       added as a close synonym to the -r option, selected for consistency with all other options in this volume of POSIX.1-2008 that do recursive
       directory descent.

       The difference between -R and the removed -r option is in the treatment by cp of file types other than regular and directory. It was imple-
       mentation-defined  how  the - option treated special files to allow both historical implementations and those that chose to support -r with
       the same abilities as -R defined by this volume of POSIX.1-2008. The original -r flag, for historic reasons, did not handle  special  files
       any  differently from regular files, but always read the file and copied its contents. This had obvious problems in the presence of special
       file types; for example, character devices, FIFOs, and sockets.
```

https://www.unix.com/man-page/posix/1p/cp/
